src/NUnitFramework/framework/Internal/TestNameGenerator.cs

```diff
@@ -0,0 +1,457 @@
+                    case "{7}":
+                    case "{8}":
+                    case "{9}":
+                        int index = token[1] - '0';
+                        _fragments.Add(new ArgumentFragment(index, 40));
+                        break;
+                    default:
+                        char c = token[1];
+                        if (token.Length >= 5 && token[2] == ':' && (c == 'a' || char.IsDigit(c)))
+                        {
+                            int length;
+
+                            // NOTE: The code would be much simpler using TryParse. However,
+                            // that method doesn't exist in the Compact Framework.
+                            try
+                            {
+                                length = int.Parse(token.Substring(3, token.Length - 4));
+                            }
+                            catch
+                            {
+                                length = -1;
+                            }
+                            if (length > 0)
+                            {
+                                if (c == 'a')
+                                    _fragments.Add(new ArgListFragment(length));
+                                else // It's a digit
+                                    _fragments.Add(new ArgumentFragment(c - '0', length));
+                                break;
+                            }
+                        }
+
+                        // Output the erroneous token to aid user in debugging
+                        _fragments.Add(new FixedTextFragment(token));
+                        break;
+                }
+
+                start = rcurly + 1;
+            }
+
+
+            // Output any trailing plain text
+            if (start < pattern.Length)
+                _fragments.Add(new FixedTextFragment(pattern.Substring(start)));
+        }
+
+        /// <summary>
+        /// Get the display name for a method
+        /// </summary>
+        /// <param name="method">A MethodInfo</param>
+        /// <returns>The display name</returns>
+        public string GetDisplayName(MethodInfo method)
+        {
+            return GetDisplayName(method, null);
+        }
+
+        /// <summary>
+        /// Get the display name for a method with args
+        /// </summary>
+        /// <param name="method">A MethodInfo</param>
+        /// <param name="args">Argument list for the method</param>
+        /// <returns>The display name</returns>
+        public string GetDisplayName(MethodInfo method, object[] args)
+        {
+            var result = new StringBuilder();
+
+            foreach (var fragment in _fragments)
+                result.Append(fragment.GetText(method, args));
+
+            return result.ToString();
+        }
+
+        #region Nested Classes Representing Name Fragments
+
+        private abstract class NameFragment
+        {
+            private const string THREE_DOTS = "...";
```

Comment:
It should be fine on any platform that supports Unicode which should be all
these days. Supported on CF?

Should be fine in XML, it is in a pretty common Unicode range.
On Oct 24, 2015 5:11 PM, "CharliePoole" <notifications@github.com> wrote:

> In src/NUnitFramework/framework/Internal/TestNameGenerator.cs
> <https://github.com/nunit/nunit/pull/912#discussion_r42938266>:
>
> > +        /// <returns>The display name</returns>
> > +        public string GetDisplayName(MethodInfo method, object[] args)
> > +        {
> > +            var result = new StringBuilder();
> > +
> > +            foreach (var fragment in _fragments)
> > +                result.Append(fragment.GetText(method, args));
> > +
> > +            return result.ToString();
> > +        }
> > +
> > +        #region Nested Classes Representing Name Fragments
> > +
> > +        private abstract class NameFragment
> > +        {
> > +            private const string THREE_DOTS = "...";
>
> The issue is whether there are platforms for on which this would not be
> acceptable. Is it OK in XML?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nunit/nunit/pull/912/files#r42938266>.
>